flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/mapper/Mapper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.mapper;
+
+import org.apache.flink.ml.api.misc.param.Params;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Collector;
+
+/**
+ * Abstract class for mappers.
+ */
+public abstract class Mapper extends FlatMapper implements MapOperable {
+
+	public Mapper(TableSchema dataSchema, Params params) {
+		super(dataSchema, params);
+	}
+
+	/**
+	 * This method override the {@link FlatMapper#flatMap(Row, Collector)} to map a row
+	 * to a new row which collected to {@link Collector}.
+	 *
+	 * @param row the input row.
+	 * @param output the output collector
+	 * @throws Exception if {@link Collector#collect(Object)} throws exception.
+	 */
+	@Override
+	public void flatMap(Row row, Collector<Row> output) throws Exception {
```

Comment:
it is same as in `ModelMapper`
I suppose this logic can be move to `FlatMapper` class directly, because in this version of code base all implementation this method is equal.

Reply:
Thanks for your advice. For One of the extended class `FlatModelMapper` not support the map() function, we can not override its flatMap() function. Thus, I can not move the function to  FlatMapper class.

Reply:
Yes, there are other good ways to implement. I'd like to choose current one. Thanks for your helpful discussion and advice!